src/Containers/Microsoft.NET.Build.Containers/AuthHandshakeMessageHandler.cs

```diff
@@ -324,6 +324,10 @@ protected override async Task<HttpResponseMessage> SendAsync(HttpRequestMessage
                 }
                 else if (response is { StatusCode: HttpStatusCode.Unauthorized } && TryParseAuthenticationInfo(response, out string? scheme, out AuthInfo? authInfo))
                 {
+                    // Load the reply so the HTTP connection becomes available to send the authentication request.
+                    // Ideally we'd call LoadIntoBufferAsync, but it has no overload that accepts a CancellationToken so we call ReadAsByteArrayAsync instead.
+                    _ = await response.Content.ReadAsByteArrayAsync(cancellationToken).ConfigureAwait(false);
```

Comment:
Nit: is it really important to have the buffering cancellable? How much data is drained typically? Cancellation will close and drop the connection. Is this something you want here?

https://github.com/dotnet/runtime/blob/82fee2692b3954ba8903fa4764f1f4e36a26341a/src/libraries/System.Net.Http/src/System/Net/Http/HttpContent.cs#L473-L475

Reply:
> Is it really important to have the buffering cancellable

It's important we drain the response while respecting the cancellation token.

> How much data is drained typically?

Not much. None to a few hundred of bytes at most.

> Cancellation will close and drop the connection. Is this something you want here?

Yes.